lms/djangoapps/completion_api/views.py

```diff
@@ -358,11 +365,67 @@ class CompletionDetailView(CompletionViewMixin, APIView):
     """
     # pylint: enable=line-too-long
 
-    def get(self, request, course_key):
+    def get(self, request, course_id):
         """
         Handler for GET requests.
         """
-        course_key = CourseKey.from_string(course_key)
-        progress = self.get_progress_queryset().get(course_id=course_key)
+        course_id = CourseKey.from_string(course_id)
+        progress = self.get_progress_queryset().get(course_id=course_id)
         completion = CourseCompletionFacade(progress)
         return Response(self.get_serializer()(completion, requested_fields=self.get_requested_fields()).data)
+
+
+class CompletionBlockUpdateView(APIView):
+    """
+    API view to mark blocks as completed.
+    This is a transitional implementation.
+    """
+
+    authentication_classes = (
+        authentication.OAuth2AuthenticationAllowInactiveUser,
+        authentication.SessionAuthenticationAllowInactiveUser
+    )
+    permission_classes = (IsAuthenticated,)
+
+    def post(self, request, course_id, usage_id):
+        # For forwards-compatibility
+        try:
+            completion = float(request.data.get('completion'))
+        except (TypeError, ValueError):
+            return Response(status=status.HTTP_400_BAD_REQUEST)
+
+        if completion != 1:
+            return Response(status=status.HTTP_400_BAD_REQUEST)
+
+        # Thanks to URL validation, course key must be formatted correctly.
+        # Check if a course exists for this key.
+        course_key = CourseKey.from_string(course_id)
+        try:
+            get_course(course_key)
+        except ValueError:
+            return Response(status=status.HTTP_404_NOT_FOUND)
+
+        # Check if the user is enrolled in this course.
+        if not CourseEnrollment.is_enrolled(request.user, course_key):
+            return Response(status=status.HTTP_404_NOT_FOUND)
+
+        # Check if content exists for this usage_id.
+        content_key = UsageKey.from_string(usage_id)
+        try:
+            modulestore().get_item(content_key)
+        except ItemNotFoundError:
+            return Response(status=status.HTTP_404_NOT_FOUND)
+
+        # Check if the content belongs to the given course
+        course_structure = CourseStructure.objects.get(course_id=course_key)
+        if usage_id not in course_structure.structure['blocks'].keys():
+            return Response(status=status.HTTP_404_NOT_FOUND)
+
+
+        _, created = CourseModuleCompletion.objects.get_or_create(
+            user_id=request.user.id,
+            course_id=course_key,
+            content_id=usage_id,
+        )
+
+        return Response(status=status.HTTP_201_CREATED)
```

Comment:
It might be better to return 201 if a new entry was created, and 200 if the entry already existed? Or just 200 in any case.